spring-cloud-stream/src/main/java/org/springframework/cloud/stream/config/BindingServiceProperties.java

```diff
@@ -127,9 +127,10 @@ public void setEnvironment(Environment environment) {
 		if (environment instanceof ConfigurableEnvironment) {
 			// override the bindings store with the environment-initializing version if in
 			// a Spring context
+			Map<String, BindingProperties> delegate = new TreeMap<String, BindingProperties>(String.CASE_INSENSITIVE_ORDER);
```

Comment:
It is observed that `EnvironmentAware` interfaces are invoked via bean post processors every time the `refreshContext` is invoked. In this case, the `bindings` would have already been set with some values.